### PR TITLE
feat(sumologicexporter, sumologicschemaprocessor)!: move translating metadata from sumologic exporter to sumologic schema processor

### LINKED ISSUE #649: Move metadata attribute translation from sumologicexporter to sumologicschemaprocessor
Currently, our exporter takes care of translating attribute names from the OpenTelemetry schema to the Sumo schema used by apps. See the definitions [here](https://github.com/SumoLogic/sumologic-otel-collector/blob/main/pkg/exporter/sumologicexporter/translate_attributes.go).

However, exporters shouldn't modify data, as this causes race conditions and memory corruption when the same records are multiplexed into more than one exporter. Originally, the intent of this feature was to make it easier for users to send data with correct metadata to Sumo without having to configure multiple components, but in hindsight, this wasn't worth the problems it causes.

We've since introduced a [separate processor](https://github.com/SumoLogic/sumologic-otel-collector/tree/main/pkg/processor/sumologicschemaprocessor) for the purpose of translating metadata to what the Sumo backend and apps expect. The attribute translation should live in this processor instead.

This would be a breaking change and the fix should come with migration instructions.

### PR DESCRIPTION
**Pull request by aboguszewski-sumo**
Fixes #649 

The `sumologic` exporter modifies data it gets by translating attributes from OpenTelemetry convention to Sumo convention. This PR moves this logic to the `sumologic schema` processor.

Technically this is ready for review, but it might have not been tested properly yet, so I'm marking this as a draft.

### CODE REVIEW

File: CHANGELOG.md
Comment (sumo-drosiek) [inline CHANGELOG.md:12]: Not sure if we want to make it breaking change or deprecate it first

File: docs/Upgrading.md
Comment (sumo-drosiek) [inline docs/Upgrading.md:3]: I would skip this header in ToC

Comment (aboguszewski-sumo) [reply docs/Upgrading.md]: An oversight of mine, this is an evildoing of a vscode extension which I have missed, I'm sorry.

File: pkg/exporter/sumologicexporter/exporter_test.go
Comment (aboguszewski-sumo) [inline pkg/exporter/sumologicexporter/exporter_test.go:410]: In my opinion, the tests removed in file `exporter_test.go` are not very important and can be removed safely (because similar tests have been added to the processor), but please check if I'm right. The reason they are so complex is the fact that this feature couldn't be tested in an easier way while it has been in the exporter.

File: docs/Upgrading.md
Comment (sumo-drosiek) [inline docs/Upgrading.md:22]:
```suggestion
### `sumologic` exporter: moved translating attributes to `sumologicschema` processor
```

File: docs/Upgrading.md
Comment (sumo-drosiek) [inline docs/Upgrading.md:50]: I think we can skip non related part and change service to:

```
processors:
  # - ...
  - sumologicschema
```

File: pkg/processor/sumologicschemaprocessor/README.md
Comment (sumo-drosiek) [inline pkg/processor/sumologicschemaprocessor/README.md:56]:
```suggestion
but Sumo Logic expects it to be in attribute named `pod`.
```

and for other occurrences the same

**Review by sumo-drosiek — CHANGES_REQUESTED** [review]
Overall LGTM, but I would like to discuss if we can go with this change as breaking change, or now as deprecation and breaking change in the future

**Comment by sumo-drosiek** [comment]
@aboguszewski-sumo please move exporter changes (actual breaking change) to another PR and set appropriate milestone, should be `v0.59`. In addition please add deprecation log to the exporter, like in https://github.com/SumoLogic/opentelemetry-collector-contrib/commit/c9d2ba5d0f04cc3460495bb7c7018fa33b141b4f

**Comment by aboguszewski-sumo** [comment]
Done. I'll create the other pull request once this gets merged, to avoid rebasing too many times. As for now, it is located here: https://github.com/aboguszewski-sumo/sumologic-otel-collector/tree/sumologic-exporter-remove-translation

File: CHANGELOG.md
Comment (sumo-drosiek) [inline CHANGELOG.md:17]: We can add it to description as well

Comment (sumo-drosiek) [reply CHANGELOG.md:17]: https://github.com/SumoLogic/sumologic-otel-collector/blob/main/CHANGELOG.md?plain=1#L21-L23

**Review by sumo-drosiek — APPROVED** [review]
Thanks :rocket:

**Pull request merged**
